src/audio_unit/looper.rs

```diff
@@ -44,20 +47,28 @@ impl Looper {
         }
     }
 
-    fn process_message(&mut self, _: Message) {
-        self.state = match self.state {
-            Off => Recording { position: 0 },
-            Recording { position } => Playing {
-                position: 0,
-                total: position,
-            },
-            Playing {
-                position: _,
-                total: _,
-            } => Off,
-        };
+    fn process_message(&mut self, message: Message) {
+        match message {
+            Message::Toggle => {
+                self.state = match self.state {
+                    Off => Recording { position: 0 },
+                    Recording { position } => Playing {
+                        position: 0,
+                        total: position,
+                    },
+                    _ => Off,
+                };
+
+                println!("looper: {:?}", self.state);
+            }
+            Message::QueueOverdub => {
+                if let Playing { position, total } = self.state {
+                    self.state = PlayingAwaitingOverdub { position, total };
 
-        println!("looper: {:?}", self.state);
+                    println!("looper: enabling overdub mode");
+                }
+            }
+        }
```

Comment:
I think this is fine as-is, but I could also imagine the `match` logic here living in an `impl State`.